python/lsst/cp/pipe/deferredCharge.py

```diff
@@ -0,0 +1,1019 @@
+        start : `int`, optional
+            First overscan column to fit. This number includes the
+            last imaging column, and needs to be adjusted by one when
+            using the overscan bounding box.
+        stop : `int`, optional
+            Last overscan column to fit. This number includes the
+            last imaging column, and needs to be adjusted by one when
+            using the overscan bounding box.
+
+        Returns
+        -------
+        res : `np.ndarray`, (nMeasurements, nCols)
+            Model results.
+        """
+        v = params.valuesdict()
+        try:
+            v['cti'] = 10**v['ctiexp']
+        except KeyError:
+            pass
+
+        # Adjust column numbering to match DM overscan bbox.
+        start += 1
+        stop += 1
+
+        x = np.arange(start, stop+1)
+        res = np.zeros((signal.shape[0], x.shape[0]))
+
+        for i, s in enumerate(signal):
+            res[i, :] = (np.minimum(v['trapsize'], s*v['scaling'])
+                         * (np.exp(1/v['emissiontime']) - 1.0)
+                         * np.exp(-x/v['emissiontime'])
+                         + s*num_transfers*v['cti']**x
+                         + v['driftscale']*s*np.exp(-x/float(v['decaytime'])))
+
+        return res
+
+
+class SimulatedModel(OverscanModel):
+    """Simulated overscan model."""
+
+    @staticmethod
+    def model_results(params, signal, num_transfers, amp, start=1, stop=10, trap_type=None):
+        """Generate a realization of the overscan model, using the specified
+        fit parameters and input signal.
+
+        Parameters
+        ----------
+        params : `lmfit.Parameters`
+            Object containing the model parameters.
+        signal : `np.ndarray`, (nMeasurements)
+            Array of image means.
+        num_transfers : `int`
+            Number of serial transfers that the charge undergoes.
+        amp : `lsst.afw.cameraGeom.Amplifier`
+            Amplifier to use for geometry information.
+        start : `int`, optional
+            First overscan column to fit. This number includes the
+            last imaging column, and needs to be adjusted by one when
+            using the overscan bounding box.
+        stop : `int`, optional
+            Last overscan column to fit. This number includes the
+            last imaging column, and needs to be adjusted by one when
+            using the overscan bounding box.
+        trap_type : `str`, optional
+            Type of trap model to use.
+
+        Returns
+        -------
+        results : `np.ndarray`, (nMeasurements, nCols)
+            Model results.
+        """
+        v = params.valuesdict()
+
+        # Adjust column numbering to match DM overscan bbox.
+        start += 1
+        stop += 1
+
+        # Electronics effect optimization
+        output_amplifier = FloatingOutputAmplifier(1.0, v['driftscale'], v['decaytime'])
+
+        # CTI optimization
+        v['cti'] = 10**v['ctiexp']
```

Comment:
Ah, I see you do the same thing again here. The whole "having one value stored in two different places" thing always makes me uneasy, especially when they're changing. I wonder if there's a safer way to do this...

Reply:
Maybe the `KeyError` example above would be less weird if it was done in the negative, like `if 'cti not in v.keys()` so that it's clear you're calculating it from the `ctiexp` if and only if it's not already there? I dunno...